### PR TITLE
Minor backslash-related changes

### PR DESCRIPTION
**Pull request by marc-h38**
Cosmetic fixes submitted without the major code change in PR #6241 just to make sure they pass CI.

### CODE REVIEW

File: mesonbuild/mparser.py [hunk 1]
```diff
@@ -20,6 +20,30 @@
...
+#   fail to compile. Other unknown sequences discard the backslash as
+#   no-op, e.g: r'\=' matches exactly like r'='.
+#
+# - A backslash is treated the SAME inside vs outside a regex set [ ].
+#   However '\+' makes a difference outside [ ]\+, whereas the same
+#   backslash is discarded as a no-op inside [\+] because + is not
+#   special inside.
+#
+# - Regular, not-raw strings combine backslashes with the next character
+#   SOMETIMES and sometimes not; check the relevant documentation. In
```
Comment (QuLogic) [inline mesonbuild/mparser.py]: Not, sometimes, always. The cases where it doesn't cause warnings and will be errors in the future.

File: test cases/common/36 tryrun/meson.build [hunk 2]
```diff
@@ -14,10 +14,7 @@ endif
 ok_code = '''#include<stdio.h>
 int main(void) {
   printf("%s\n", "stdout");
-  fprintf(stderr, "%s\n", "stderr");
-  return 0;
-}
-'''
+''' + 'fprintf(stderr, "%s\\n", "stderr"); return 0; }'
```
Comment (QuLogic) [inline test cases/common/36 tryrun/meson.build]: I don't understand why this change is necessary.

Comment (marc-h38) [reply test cases/common/36 tryrun/meson.build]: The current description is:
> _Cosmetic_ fixes submitted without the major code change in PR #6241 just to make sure they pass CI.

File: mesonbuild/mparser.py (hunk 1 above)
Comment (QuLogic) [reply mesonbuild/mparser.py]: Read the next paragraph:

>    Changed in version 3.6: Unrecognized escape sequences produce a `DeprecationWarning`. In some future version of Python they will be a `SyntaxError`.

You should not encourage unescaped backslashes.

I don't know why you mention Meson's syntax when this comment is about Python.

File: test cases/common/36 tryrun/meson.build (hunk 2 above)
Comment (QuLogic) [reply test cases/common/36 tryrun/meson.build]: Cosmetically, it looks worse, so that doesn't explain why.

File: mesonbuild/mparser.py (hunk 1 above)
Comment (marc-h38) [reply mesonbuild/mparser.py]: > You should not encourage unescaped backslashes.

Good thing I'm not :-)

> I don't know why you mention Meson's syntax when this comment is about Python.

Various reasons but I'd rather not "review the review". In fact I didn't expect this particular PR to be reviewed at all, see its description and draft status. Thanks anyway!

File: test cases/common/36 tryrun/meson.build (hunk 2 above)
Comment (marc-h38) [reply test cases/common/36 tryrun/meson.build]: Should I have copied the commit message in the description? That seems to be common practice on github which treats commit messages poorly.

File: mesonbuild/mparser.py (hunk 1 above)
Comment (marc-h38) [reply mesonbuild/mparser.py]: BTW this does not show any `DeprecationWarning`, any idea why?
```
printf '%s\n' "s='a\zb'; print(s)" > unknownescape.py
python3.7 unknownescape.py 
python3.7 -m compileall .
```

**Comment by krader1961** [comment]
Someone please close this. The author clearly did not mean for it to be merged. Also, why is it necessary to create a PR to run CI tests? The ksh project makes it easy for people to run Travis CI tests on their own private fork prior to creating a public PR. Isn't that true for this project?

**Comment by marc-h38** [comment]
Curious why you waste your time looking at obscure drafts? There are 157 draft:false PRs open right now.

> The ksh project makes it easy for people to run Travis CI tests on their own private fork prior to creating a public PR. Isn't that true for this project?

Maybe no one found the time to set it up or to document it? Maybe you could help...
